### DISCUSSION TITLE
CommandPalete\NewTab... Type a command name ... allow exe name

### DISCUSSION THREAD

**Discussion by vefatica** [microsoft/terminal#13997] [Ideas]
In order to help a user, I downloaded and installed an ancient version of TCC.  I figured the CommandPalette\NewTab... was the perfect place to run it.  Nope!  The only thing accepted in the "Type a command name" box is the name of something already appearing in the list of profiles below (which you can pretty easily get to with a few down-arrows).  That seems a little lame.  Can you allow a fully-qualified name of a console executable in that box (so it would work like "wt nt drive:\path\app.exe")?

Please tell me if I've missed a way to NewTab an un-profiled app.

Ditto for the CommandPalette\SplitPane... dialog.

**Comment by DHowett** [comment]
This is a great question! By default, the command palette lets you run any individual pre-defined action from your settings. _However!_ If you backspace the `>` at the beginning of the prompt, you can run "wt" commandline commands. This includes spawning new tabs hosting applications (and it's pretty easy, too!)

<img width="179" alt="image" src="https://user-images.githubusercontent.com/189190/190214420-729981b6-99fa-4ccd-9830-0570a624c72d.png">

Backspace:

<img width="187" alt="image" src="https://user-images.githubusercontent.com/189190/190214445-18807238-1bd5-45ae-80bf-b097f0d48fe8.png">

Type something:

<img width="410" alt="image" src="https://user-images.githubusercontent.com/189190/190214503-35e892fa-6690-4457-b596-d19fe50b5c2c.png">

Observe that it can even take multiple commands and tell you what they do:

<img width="359" alt="image" src="https://user-images.githubusercontent.com/189190/190214646-995158a5-a65c-4439-88e8-9ebafca2e1b6.png">

**Reply by vefatica** [reply]
That's certainly easy enough!  But who would have known?  Maybe the prompt could be changed to

`Type a profile name or backspace to enter a command`

I did the same in the CommandPalette\SplitPane... dialog and got a new pane!  Bug?  It's nice that those two dialogs have a shared history.

**Reply by DHowett** [reply]
You're right, we could surface this feature a lot better.

Interesting about the nested command issue. @zadjii-msft, should we suppress backspace=commandline mode when you're in a nested command? We need to refine the UX for nested commands anyway...

_Idle thought: what if entering commandline mode while in a nested command could generate a partial command for you, so inside "split pane..." backspace would give you `sp`. It would let you upgrade/downgrade actions into commandlines...!_

**Reply by vefatica** [reply]
P.S., I didn't include "nt" in the NewPane dialog, just the exe name that was OK (got a new pane).  Given that the two dialogs have different names, couldn't the "nt" and "sp" be automatic?

It's a bit weird that if you put "sp" in the NewPane dialog you get a split pane and if you put "nt" (or nothing) in the SplitPane dialog you get a new pane.

**Reply by zadjii-msft** [reply]
> should we suppress backspace=commandline

this seems doable

> Idle thought: what if entering commandline mode while in a nested command could generate a partial command

This seems less doable. Next to impossible, I'd reckon.

**Reply by vefatica** [reply]
The backspace is fine but it needs to be in the prompt.  "nt" and "sp" should be the defaults in their respective dialogs.

**Reply by vefatica** [reply]
How about key actions to get to the NewTab and SplitPane dialogs, in addition to `>`?  Alt+shift+n and alt+shift+s seem to be available.
